ts/Data/Modifiers/ChainModifier.ts

```diff
@@ -151,6 +151,56 @@ class ChainModifier extends DataModifier {
         });
     }
 
+    /**
+     * Applies several modifications to the table and returns a modified copy of
+     * the given table.
+     *
+     * @param {Highcharts.DataTable} table
+     * Table to modify.
+     *
+     * @param {DataEvent.Detail} [eventDetail]
+     * Custom information for pending events.
+     *
+     * @return {Promise<Highcharts.DataTable>}
+     * Table with `modified` property as a reference.
+     */
+    public modify<T extends DataTable>(
+        table: T,
+        eventDetail?: DataEvent.Detail
+    ): Promise<T> {
+        const chain = this,
+            modifiers = (
+                chain.options.reverse ?
+                    chain.modifiers.reverse() :
+                    chain.modifiers.slice()
+            );
+
+        let promiseChain: Promise<T> = Promise.resolve(table);
+
+        for (let i = 0, iEnd = modifiers.length; i < iEnd; ++i) {
+            const modifier = modifiers[i];
+            promiseChain = promiseChain.then((chainTable): Promise<T> =>
+                modifier.modify(chainTable.modified as T, eventDetail)
+            );
+        }
+
+        promiseChain = promiseChain.then((chainTable): T => {
+            table.modified = chainTable.modified;
+            return table;
+        });
+
+        promiseChain = promiseChain['catch']((error): Promise<T> => {
```

Comment:
I might be missing something, but could this not be a `.catch` on the assignment above?

Reply:
Kind of, but I have to use brackets for IE and then I can not make proper alignments and it becomes ugly code. So this is an aesthetical thing to make the code more readable.

Reply:
Ah, IE 😢